app/controllers/api/v1/npq_applications_controller.rb

```diff
@@ -26,19 +26,15 @@ def show
       end
 
       def reject
-        if NPQ::Reject.call(npq_application:)
-          render json: json_serializer_class.new(npq_application).serializable_hash
-        else
-          render json: { errors: Api::ErrorFactory.new(model: npq_application).call }, status: :bad_request
-        end
+        service = RejectNPQApplication.new(npq_application:)
+
+        render_from_service(service, json_serializer_class)
       end
 
       def accept
-        if NPQ::Accept.call(npq_application:)
-          render json: json_serializer_class.new(npq_application).serializable_hash
-        else
-          render json: { errors: Api::ErrorFactory.new(model: npq_application).call }, status: :bad_request
-        end
+        service = AcceptNPQApplication.new(npq_application:)
```

Comment:
can we retain the `NPQ::AcceptApplication` structure, as this only relates to NPQ applications and no ECF? same for the reject, what do you think?

Reply:
Hi @cwrw tks for your review. Addressed in https://github.com/DFE-Digital/early-careers-framework/pull/2674/commits/8c9fde545669f063b8a66cd2c524248a8025fdb5